src/sentry/integrations/slack/utils/notifications.py

```diff
@@ -198,6 +304,108 @@ def _save_notification_message(
         pass
 
 
+def _save_notification_message_notification_action(
+    notification_message_object: NewNotificationActionNotificationMessage,
+    repository: NotificationActionNotificationMessageRepository,
+) -> None:
+    try:
+        repository.create_notification_message(data=notification_message_object)
+    except Exception:
+        pass
+
+
+def _handle_workflow_engine_notification(
+    organization: Organization,
+    notification_context: NotificationContext,
+    metric_issue_context: MetricIssueContext,
+    integration: RpcIntegration,
+    attachments: str,
+    text: str,
+    channel: str,
+) -> bool:
+    try:
+        group = Group.objects.get(id=metric_issue_context.id)
+    except Group.DoesNotExist:
+        _logger.info(
+            "Group not found",
+            extra={"incident_id": metric_issue_context.id},
+        )
+        return False
+
+    open_period_start = open_period_start_for_group(group)
```

Comment:
Is this concern eventually being hoisted into the caller? I don't think we want to couple group/open period logic with the workflow notification sending logic if possible.